File: moseoh/_1주차/_2JadenCase문자열만들기/java/Solution.java
package _1주차._2JadenCase문자열만들기.java;

import java.util.StringTokenizer;

public class Solution {
    public static void main(String[] args) {
        String s = "3people unFollowed me  asd  ";
        System.out.println(solution(s));
    }

    public static String solution(String s) {
        StringTokenizer st = new StringTokenizer(s, " ", true);
        StringBuilder sb = new StringBuilder();
        while (st.hasMoreTokens()) {
            sb.append(capitalize(st.nextToken()));
        }
        return sb.toString();
    }

    private static String capitalize(String s) {
        if (s.length() < 1) return s;
        else return s.substring(0, 1).toUpperCase() + s.substring(1).toLowerCase();
    }
}

File: moseoh/_1주차/_2JadenCase문자열만들기/kotlin/Solution.kt
package _1주차._2JadenCase문자열만들기.kotlin


fun main(args: Array<String>) {
    val s = "3people unFollowed me  asd  "
    println(Solution().solution(s))
}

class Solution {
    fun solution(s: String): String =
        s.split(" ").joinToString(" ") {
            it.lowercase().replaceFirstChar { c -> c.titlecase() }
        }
}